jni/src/nmslib_wrapper.cpp

```diff
@@ -27,301 +27,354 @@
 
-    // Parse query params
-    std::vector<std::string> queryParams;
+  // Parse query params
+  std::vector<std::string> queryParams;
 
-    if(parametersCpp.find("efSearch") != parametersCpp.end()) {
-        auto efSearch = std::to_string(jniUtil->ConvertJavaObjectToCppInteger(env, parametersCpp["efSearch"]));
-        queryParams.push_back("efSearch=" + efSearch);
-    }
+  auto it = parametersCpp.find("efSearch");
+  if (it != parametersCpp.end()) {
+    auto efSearch = std::to_string(jniUtil->ConvertJavaObjectToCppInteger(env, it->second));
+    queryParams.push_back("efSearch=" + efSearch);
+  }
 
-    // Load index
-    knn_jni::nmslib_wrapper::IndexWrapper * indexWrapper;
-    try {
-        indexWrapper = new knn_jni::nmslib_wrapper::IndexWrapper(spaceTypeCpp);
-        indexWrapper->index->LoadIndex(indexPathCpp);
-        indexWrapper->index->SetQueryTimeParams(similarity::AnyParams(queryParams));
-    } catch (...) {
-        delete indexWrapper;
-        throw;
+  // Load index
+  knn_jni::nmslib_wrapper::IndexWrapper *indexWrapper = nullptr;
+  try {
+    indexWrapper = new knn_jni::nmslib_wrapper::IndexWrapper(spaceTypeCpp);
+    indexWrapper->index->LoadIndex(indexPathCpp);
+    indexWrapper->index->SetQueryTimeParams(similarity::AnyParams(queryParams));
+  } catch (...) {
+    delete indexWrapper;
+    throw;
+  }
+
+  return (jlong) indexWrapper;
+}
+
+jlong knn_jni::nmslib_wrapper::LoadIndexWithStream(knn_jni::JNIUtilInterface *jniUtil,
+                                                   JNIEnv *env,
+                                                   std::istream &input,
+                                                   jobject parametersJ) {
+  if (parametersJ == nullptr) {
+    throw std::runtime_error("Parameters cannot be null");
+  }
+
+  auto parametersCpp = jniUtil->ConvertJavaMapToCppMap(env, parametersJ);
+
+  // Get space type for this index
+  jobject spaceTypeJ = knn_jni::GetJObjectFromMapOrThrow(parametersCpp, knn_jni::SPACE_TYPE);
+  std::string spaceTypeCpp(jniUtil->ConvertJavaObjectToCppString(env, spaceTypeJ));
+  spaceTypeCpp = TranslateSpaceType(spaceTypeCpp);
+
+  // Parse query params
+  std::vector<std::string> queryParams;
+
+  auto it = parametersCpp.find("efSearch");
+  if (it != parametersCpp.end()) {
+    auto efSearch = std::to_string(jniUtil->ConvertJavaObjectToCppInteger(env, it->second));
+    queryParams.push_back("efSearch=" + efSearch);
+  }
+
+  // Load index
+  knn_jni::nmslib_wrapper::IndexWrapper *indexWrapper = nullptr;
+  try {
+    indexWrapper = new knn_jni::nmslib_wrapper::IndexWrapper(spaceTypeCpp);
+    indexWrapper->index->SetQueryTimeParams(similarity::AnyParams(queryParams));
+
+    if (auto hnswFloatIndex = dynamic_cast<similarity::Hnsw<float> *>(indexWrapper->index.get())) {
+      hnswFloatIndex->LoadIndexWithStream(input);
+    } else if (auto hnswIntIndex = dynamic_cast<similarity::Hnsw<int> *>(indexWrapper->index.get())) {
```

Comment:
Currently, we dont support hnswInt. Can we remove this?

Reply:
sure, will update it.